### PR TITLE
Add hypothesis opener viewmodel with new button variant

### PR DESCRIPTION
**Pull request by davidcmoulton**
Accompanies https://github.com/elifesciences/pattern-library/pull/787

(https://github.com/elifesciences/pattern-library/pull/787 should be merged before this one.)

### CODE REVIEW

File: tests/src/ViewModel/ButtonSpeechBubbleTest.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+final class ButtonSpeechBubbleTest extends ViewModelTest
+{
+    /**
+     * @test
+     */
+    public function it_has_data()
+    {
+        $data = [
+            'text' => '<span aria-hidden="true">3</span><span class="visuallyhidden">Open annotations (there are currently 3 annotations on this page).</span>',
```
Comment (thewilkybarkid) [inline tests/src/ViewModel/ButtonSpeechBubbleTest.php]: Looking like it's be useful to split this into two properties, or would `<span class="visuallyhidden">Open annotations (there are currently</span> 3 <span class="visuallyhidden">annotations on this page).</span>` work?

Comment (thewilkybarkid) [reply tests/src/ViewModel/ButtonSpeechBubbleTest.php]: Ah no, needs the symbol instead of 0.

Comment (thewilkybarkid) [reply tests/src/ViewModel/ButtonSpeechBubbleTest.php]: Could add a new function like `mixed_visibility_text()` instead? Not sure what to call it.

File: resources/definitions/hypothesis-opener.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    js: []
+schema:
+    $schema: 'http://json-schema.org/draft-04/schema#'
+    type: object
+    properties:
+        button:
+            $schema: 'http://json-schema.org/draft-04/schema#'
+            type: object
+            allOf:
+                -
```
Comment (giorgiosironi) [inline resources/definitions/hypothesis-opener.yaml]: don't know the conventions here, but is the newline after `-` needed? e.g. https://github.com/elifesciences/api-raml/blob/8972b5fcfc67331a332319e459b4073249660c97/src/model/community-list.v1.yaml#L28-L34

File: src/ViewModel/Button.php [hunk 3]
```diff
@@ -103,6 +106,23 @@ public static function link(
...
 
+    public static function speechBubble(
+        string $text,
+        bool $isActive = true,
+        string $name = null,
+        string $id = null,
+        bool $isPopulated = false
+    ) : Button {
+        $button = new static($text, self::SIZE_CUSTOM, self::STYLE_SPEECH_BUBBLE, $isActive, $name, $id, false);
+        $button->type = self::TYPE_BUTTON;
```
Comment (giorgiosironi) [inline src/ViewModel/Button.php]: strange that the constructor doesn't set `$this->type`, any reason for the difference wrt `$name` and `$id`?

File: src/ViewModel/HypothesisOpener.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+{
+    use ArrayAccessFromProperties;
+    use ArrayFromProperties;
+    use SimplifyAssets;
+
+    private $button;
+
+    public function __construct()
+    {
+        $visibleAnnotationCount = '<span data-visible-annotation-count>&#8220;</span>';
```
Comment (giorgiosironi) [inline src/ViewModel/HypothesisOpener.php]: left double quotation mark? Whatever the character is, the magic number can be extracted into a constant

File: src/ViewModel/Button.php (hunk 3 above)
Comment (davidcmoulton) [reply src/ViewModel/Button.php]: `speechBubble` is reusing the convention already established with e.g. `form`, see https://github.com/elifesciences/patterns-php/pull/524/files/8a41c5346813bb7d5e4585a2fac9dabe441ff09c#diff-151b077966c70a0674bf1be5c891636dL72

File: resources/definitions/hypothesis-opener.yaml (hunk 2 above)
Comment (davidcmoulton) [reply resources/definitions/hypothesis-opener.yaml]: This file is an auto-generated, expanded version the equivalent file brought over from file in `pattern-library`. In `pattern-library`, this is the current convention. Happy to change it to harmonise across repos, but suggest this is out of scope for this PR.

Comment (giorgiosironi) [reply resources/definitions/hypothesis-opener.yaml]: Yes, follow the established convention (which I don't know around here)

File: tests/src/ViewModel/ButtonSpeechBubbleTest.php (hunk 1 above)
Comment (davidcmoulton) [reply tests/src/ViewModel/ButtonSpeechBubbleTest.php]: See `mixed_accessibility_text()` https://github.com/elifesciences/patterns-php/pull/524/files/8a41c5346813bb7d5e4585a2fac9dabe441ff09c#diff-84b928a9905378810b602c8a740b4cbaR58

Keeping explicit expected text in test for documentation.

**Pull request merged**